hydra-node/src/Hydra/API/Server.hs

```diff
@@ -59,7 +59,7 @@ type ServerComponent tx m a = ServerCallback tx m -> (Server tx m -> m a) -> m a
 
 withAPIServer ::
   forall tx.
-  IsChainState tx =>
+  (IsChainState tx, HasTextEnvelope tx) =>
```

Comment:
Should: avoid this cardano-specific constraint

This is oddly specific and a `ToJSON tx` should be enough?